src/components/protocol_handler/include/protocol_handler/protocol_handler_impl.h

```diff
@@ -569,6 +620,8 @@ class ProtocolHandlerImpl
 #ifdef TELEMETRY_MONITOR
   PHTelemetryObserver* metric_observer_;
 #endif  // TELEMETRY_MONITOR
+  PTUState ptu_state_;
+  struct SessionInfo pending_session_;
```

Comment:
@tkireva Protocol handler is common for all connections, so IMO ptu_state_ and pending_session_ should be combined and associated with particular connection, am I right?